.github/workflows/ros2.yml

```
@@ -0,0 +1,45 @@
+name: ros2
+
+on:
+  push:
+  pull_request:
+
+env:
+  BUILD_TYPE: Release
+
+jobs:
+  build:
+    name: Build on ros2 ${{ matrix.ros_distro }}
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        ros_distro: [foxy, galactic]
+
+    steps:   
+    - uses: ros-tooling/setup-ros@v0.2
+      with:
+        required-ros-distributions: ${{ matrix.ros_distro }}
+    
+    - name: Setup ros2 workspace
+      run: | 
+        source /opt/ros/${{ matrix.ros_distro }}/setup.bash
+        mkdir -p ${{github.workspace}}/ros2_ws/src
+        cd ${{github.workspace}}/ros2_ws
+        colcon build
+        
+    - uses: actions/checkout@v2
+      with:
+        repository: '107-systems/l3xz_openmv_camera'
+        path: 'ros2_ws/src/l3xz_openmv_camera'
+        
+    - uses: actions/checkout@v2
```

```suggestion
    - uses: actions/checkout@v3
```
Unless there is a specific reason for using the older action version, it is best to start with the up to date version.

https://github.com/actions/checkout/releases/tag/v3

---

@per1234 thanks for Your hints, I missed those points :+1: